management_api_app/db/repositories/resource_specs.py

```diff
@@ -0,0 +1,37 @@
+from typing import List
+
+from azure.cosmos import ContainerProxy, CosmosClient
+
+from models.domain.resource import ResourceSpec
+from db.repositories.base import BaseRepository
+from core.config import STATE_STORE_RESOURCE_SPECS_CONTAINER
+
+
+class ResourceSpecRepository(BaseRepository):
+    def __init__(self, client: CosmosClient):
+        super().__init__(client, STATE_STORE_RESOURCE_SPECS_CONTAINER)
+
+    @property
+    def container(self) -> ContainerProxy:
+        return self._container
+
+    def _query(self, query: str):
+        resource_specs = list(self.container.query_items(query=query, enable_cross_partition_query=True))
+        return resource_specs
+
+    def get_by_name(self, name: str, latest=False) -> List[ResourceSpec]:
+        by_name = f"SELECT * FROM ResourceSpecs rs where rs.name = {name}"
```

Comment:
Can we use the query builder?

Reply:
We could but isn't it easier to read this way right now? My intention is to make it easier to read so I suggest query builder for you since you had default clause and it would have to be remembered. Here the two query are different and sure I could refactor a bit more but I thought that comes at the expense of reading the code.

Reply:
you were the one who wanted the query builder :) I had added the queries to the repo code -- the only reason I created a query builder was because you requested it :D

Reply:
AFAIC I think we should remove the query builder everywhere - it is overengineering - so let's skip it in all repositories.... I'll undo my work in the next PR